### PR TITLE
XW-3899 | update featured video labels in custom dimension

### PR DESCRIPTION
**Pull request by latata**
https://wikia-inc.atlassian.net/browse/XW-3899
@Wikia/x-wing 
@Wikia/adeng I added additional null check because there was exception thrown on browsers with adblock enabled
https://github.com/Wikia/app/compare/XW-3899?expand=1#diff-9d6fc6ece7c9753b14648b1f8b3c4870R117

### CODE REVIEW

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 1]
```diff
@@ -367,8 +374,15 @@ require([
...
 				ooyalaVideoController.updateAdSet(ooyalaAdSetProvider.get(recommendedVideoDepth + 1, correlator, {
 					contentSourceId: videoData.dfpContentSourceId,
-					videoId: nextVideoData.embed_code
+					videoId: nextVideoId
 				}));
+				// we need to hold tracking until we update custom dimensions with next video data
+				trackingQueue.hold();
+				ooyalaService.getLabels(nextVideoId).then(function (labels) {
+					videoLabels = labels.join(',');
+					updateVideoCustomDimensions(nextVideoData.embed_code, nextVideoData.name, videoLabels);
```
Comment (xkxd) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: `nextVideoData.embed_code` => `nextVideoId`?

**Review by xkxd — APPROVED** [review]

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 2]
```diff
@@ -222,9 +227,13 @@ require([
 			});
 		}
 
-		window.guaSetCustomDimension(34, videoId);
-		window.guaSetCustomDimension(35, videoTitle);
-		window.guaSetCustomDimension(36, videoLabels);
+		function updateVideoCustomDimensions(videoId, videoTitle, videoLabels) {
```
Comment (hakubo) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: can this funciton be inited outside initAttributionTracking?

File: extensions/wikia/ArticleVideo/scripts/ooyala-service.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+define('wikia.articleVideo.ooyalaService', ['wikia.nirvana'], function (nirvana) {
+	return {
+		getLabels: function (videoId) {
+			var deferred = $.Deferred();
+			nirvana.sendRequest({
```
Comment (hakubo) [inline extensions/wikia/ArticleVideo/scripts/ooyala-service.js]: sendRequest returns $.ajax - that already returns jquery promise

**Review by hakubo — CHANGES_REQUESTED** [review]

File: extensions/wikia/ArticleVideo/ArticleVideoController.class.php [hunk 4]
```diff
@@ -29,4 +29,15 @@ public function related() {
 			$this->skipRendering();
 		}
 	}
+
+	public function labels() {
+		$videoId = $this->getVal( 'videoId', null );
+		if ( empty( $videoId ) ) {
```
Comment (hakubo) [inline extensions/wikia/ArticleVideo/ArticleVideoController.class.php]: one \n before if

**Review by hakubo — APPROVED** [review]

File: extensions/wikia/ArticleVideo/ArticleVideoController.class.php [hunk 5]
```diff
@@ -29,4 +29,16 @@ public function related() {
...
+	public function labels() {
+		$videoId = $this->getVal( 'videoId', null );
+
+		if ( empty( $videoId ) ) {
+			throw new MissingParameterApiException( 'videoId' );
+		}
+
+		$api = OoyalaBacklotApiService::getInstance();
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+		$this->response->setVal( 'labels', $api->getLabels( $videoId ) );
```
Comment (kvas-damian) [inline extensions/wikia/ArticleVideo/ArticleVideoController.class.php:42]: It's a request to 3rd party API. Can we handle errors and make sure site won't be down when there are any problems on Ooyala side?

**Comment by kvas-damian** [comment]
What is the use-case for this data?
I'm not sure how would we like to use this data. We don't share this information in the user session as it's changing. This suggests me that we're tracking one-time data - event. Maybe we could use tracking events?

**Pull request merged**